### DISCUSSION TITLE
Example for advection + source term in 3D?

### DISCUSSION THREAD

**Discussion by md-schroer** [usnistgov/fipy#1081] [Q&A]
Hey, I'm looking to use fipy for a little modeling task I'm working on.  What I'd like to do is load a few numpy arrays, which give the velocity (vx,  vy, vz) on a 3D rectilinear grid.  Then load another few arrays that define some other scalar properties over the grid (think temperature, intensity etc).  Finally, solve for the steady state taking into account advection and a source term (which is a function of our scalar parameters, but is independent of time).

Is there a good example for how to get started with this with fipy? I was having a hard time finding something similar.

Thanks!

**Comment by wd15** [comment]
I think FiPy can handle that without too many issues depending on the size of the grid.  Probably start with the [Diffusion Examples](https://www.ctcms.nist.gov/fipy/examples/README.html) and then move onto the [Convection Examples](https://www.ctcms.nist.gov/fipy/examples/convection/index.html). It seems like FiPy is missing some good 2D/3D convection examples, but the general concepts should be clear from a mix of the provided examples.
